domain/service/servicehealth_store.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2016 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"strings"
+	"errors"
+
+	"github.com/control-center/serviced/datastore"
+	"github.com/control-center/serviced/datastore/elastic"
+)
+
+func (s *storeImpl) GetServiceHealth(ctx datastore.Context, svcId string) (*ServiceHealth, error) {
+	defer ctx.Metrics().Stop(ctx.Metrics().Start("storeImpl.GetServiceHealth"))
+
+	id := strings.TrimSpace(svcId)
+	if id == "" {
+		return nil, errors.New("empty service id not allowed")
+	}
+
+	searchRequest := newServiceHealthElasticRequest(map[string]interface{}{
+		"query": map[string]interface{}{
+			"ids": map[string]interface{}{
+				"values": []string{id},
+			},
+		},
+		"fields": serviceHealthFields,
+		"size":   1,
+	})
+
+	results, err := datastore.NewQuery(ctx).Execute(searchRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	if results.HasNext() {
+		var health ServiceHealth
+		err = results.Next(&health)
+		if err != nil {
+			return nil, err
+		}
+		s.fillHealthVolatileInfo(&health)
+		return &health, nil
+	}
+
+	// no errors but also no results for given id
```

Comment:
For consistency with the similar methods on this class, this case should return `ErrNoSuchEntity`. See the latest version of `GetServiceDetails()` on develop for an example.